src/controllers/queryRunner.ts

```diff
@@ -1681,6 +1671,31 @@ export default class QueryRunner {
         return value;
     }
 
+    /**
+     * Vscode core expects uri.fsPath for resourcePath context value.
+     * https://github.com/microsoft/vscode/blob/bb5a3c607b14787009f8e9fadb720beee596133c/src/vs/workbench/common/contextkeys.ts#L275
+     */
+
+    /**
+     * Add query to running queries list
+     * @param ownerUri The owner URI of the query
+     */
+    private addRunningQuery(ownerUri: string): void {
+        const key = vscode.Uri.parse(ownerUri).fsPath;
+        QueryRunner._runningQueries.push(key);
+        this.updateRunningQueries();
+    }
+
+    /**
+     * Remove current query from running queries list
+     */
+    private removeRunningQuery(): void {
+        QueryRunner._runningQueries = QueryRunner._runningQueries.filter(
```

Comment:
It'd odd to me that we're storing _runningQueries as an array (rather than Set) and _connections as a record (rather than a Map).  Not saying we have to swap those now, but probably something for the future so that we can get the unqiueness guarantees that we seem to be expecting for those.

Then we can just serialize those into more primitive forms when we update the context values.